### ISSUE TITLE
Disabling and re-enabling the contacts app wipes all contacts users had previously

### ISSUE THREAD

**Issue by danry25** [nextcloud/contacts#714]
### Describe the bug
I disabled this app in Nextcloud and re-enabled it. It appears all of my contacts are now gone from Nextcloud.

**To Reproduce**
1. Go to Apps => Active apps
2. Click on Disable to the right of Contacts
3. Go to Apps => Disabled apps
4. Click on Enable to the right of Contacts

**Expected behavior**
User data should be retained despite this app being disabled & summarily enabled. One doesn't disable something then expect it to be a clean slate when they re-enable it.

**Actual behavior**
All user data is lost without notification or warning, hopefully you had good backups of your contacts!

### Server configuration
**Operating system**: openSUSE Tumbleweed Linux 4.17.6-1-default #1 SMP PREEMPT Wed Jul 11 20:01:45 UTC 2018 (8a8b6ec) x86_64

**Web server:** nginx/1.15.3 (fpm-fcgi)

**Database:** mysql 10.2.18

**PHP version:** 7.2.10
Modules loaded: Core, date, libxml, pcre, filter, hash, Reflection, SPL, session, SimpleXML, standard, xml, mysqlnd, cgi-fcgi, bz2, ctype, curl, dom, fileinfo, gd, gettext, iconv, imagick, json, mbstring, mysqli, openssl, PDO, pdo_mysql, pdo_sqlite, posix, redis, sqlite3, tokenizer, xmlreader, xmlwriter, zip, zlib

**Nextcloud version:** 14.0.3 - 14.0.3.0

**List of Activated Apps** Enabled:
 - admin_notifications: 1.0.2
 - calendar: 1.6.3
 - cloud_federation_api: 0.0.1
 - contacts: 2.1.6
 - dav: 1.6.0
 - encryption: 2.2.0
 - federatedfilesharing: 1.4.0
 - files: 1.9.0
 - files_pdfviewer: 1.3.2
 - files_rightclick: 0.8.4
 - files_sharing: 1.6.2
 - files_texteditor: 2.6.0
 - files_trashbin: 1.4.1
 - files_versions: 1.7.1
 - firstrunwizard: 2.3.0
 - gallery: 18.1.0
 - issuetemplate: 0.4.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.2.0
 - notifications: 2.2.1
 - oauth2: 1.2.1
 - password_policy: 1.4.0
 - provisioning_api: 1.4.0
 - qownnotesapi: 18.8.0
 - serverinfo: 1.4.0
 - sharebymail: 1.4.0
 - support: 1.0.0
 - theming: 1.5.0

**Updated from an older Nextcloud or fresh install:** Updated from older version

**Signing status:** Looks like an empty array?

**Nextcloud configuration:**

```
{
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "localhost",
        "aidanis.online"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "14.0.3.0",
    "overwrite.cli.url": "http:\/\/localhost",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpmode": "smtp",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauthtype": "PLAIN",
    "mail_smtpauth": 1,
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpsecure": "tls",
    "mail_smtpport": "587",
    "maintenance": false,
    "theme": "",
    "logfile": "\/var\/log\/nextcloud.log",
    "loglevel": 2,
    "memcache.local": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 6379
    }
}
```

### Client configuration
**Browser:** Firefox 60.3.0esr (64-bit)

**Operating system:** Debian Stretch

**CardDAV-clients:** Web client?

**Comment by skjnldsv** [comment]
Hello! :) 
This is not possible.
The data and data synchronisation have nothing to do with the contacts app.
The contacts app is just an interface to edit your already synced contacts. Do you have access to your database?

**Comment by danry25** [comment]
I do have access to the database, any pointers on where to poke around in it or things to try?

**Comment by skjnldsv** [comment]
Yes, you need to check the oc_cards table. All your contacts should be there. :)

**Comment by skjnldsv** [comment]
@danry25 good news :+1: :)
